src/Compilers/Test/Core/CollectionBuilderAttribute.cs

```diff
@@ -0,0 +1,23 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#if !NET8_0_OR_GREATER
```

Comment:
Just to make sure that we don't run afoul of any weird situations (such as razor test dependencies), we should typeforward in an `#else`: https://github.com/dotnet/roslyn/blob/main/docs/Target%20Framework%20Strategy.md#pattern-for-types

Reply:
Thanks for the pointer. That's a useful doc